test/spec/GarnSpec.hs

```diff
@@ -60,67 +61,69 @@ spec = do
                   init
               |]
 
-    -- TODO: Golden tests currently can’t be integrated with the other test cases
-    --       because stackbuilders/hspec-golden#40. The case below shows the
-    --       effect that @`around_` `inTempDirectory`@ _should_ have.
-    describe "garn-golden" $ do
-      describe "run" $ do
-        it "generates formatted flakes" $ do
-          inTempDirectory $ do
-            writeHaskellProject repoDir
-            _ <- runGarn ["run", "foo"] "" repoDir Nothing
-            flake <- readFile "./flake.nix"
-            pure $ defaultGolden "generates_formatted_flakes" flake
+      it "generates formatted flakes" $ do
+        inTempDirectory $ do
+          writeHaskellProject repoDir
+          _ <- runGarn ["run", "foo"] "" repoDir Nothing
+          flake <- readFile "./flake.nix"
+          pure $ defaultGolden "generates_formatted_flakes" flake
 
-    describe "version mismatches" $ around onTestFailureLogger $ do
-      it "gives a helpful error messages when there's a deno <-> haskell json version mismatch" $
-        \onTestFailureLog -> do
-          inTempDirectory $ do
-            writeFile
-              "garn.ts"
-              [i|
-                type DenoOutput = {
-                  "some incompatible type": null,
-                  garnTsLibVersion: string
-                };
+      it "outputs a version with --version" $ onTestFailureLogger $ \onTestFailureLog -> do
+        output <- runGarn ["--version"] "" repoDir Nothing
+        onTestFailureLog output
+        stdout output `shouldBe` "v0.0.14\n"
+        stderr output `shouldBe` ""
+        exitCode output `shouldBe` ExitSuccess
```

Comment:
The diff for this file looks terrible. It's probably easier to review with hidden white space changes. Anyways, this is the only real now test.